### PR TITLE
More tests and refactoring

### PR DESCRIPTION
**Pull request by madadam**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -736,10 +734,15 @@ mod test {
             assert!(test.table.add_node(test.node_info.clone()).is_some());
         }
 
+        let prev_node_count = test.table.len();
```
Comment (afck) [inline src/lib.rs]: Shouldn't that be exactly `GROUP_SIZE`?

File: src/lib.rs [hunk 2]
```diff
@@ -736,10 +734,15 @@ mod test {
...
+
         test.node_info.public_id.set_name(get_contact(&test.name, 1, 255));
         assert_eq!(test.table.add_node(test.node_info.clone()),
                    Some((Vec::new(), true)));
 
+        // Adding a node should not remove exiting nodes
```
Comment (afck) [inline src/lib.rs]: "existing" (also below)

File: src/lib.rs [hunk 3]
```diff
@@ -942,14 +957,8 @@ mod test {
...
-            let mut assert_checker = 0;
-            for j in 0..target_nodes.len() {
-                if *target_nodes[j].name() == get_contact(&test.name, i, 1) {
-                    assert_checker = 1;
-                    break;
-                }
-            }
-            assert!(assert_checker == 1);
+            let expected_name = get_contact(&test.name, i, 1);
+            assert!(target_nodes.iter().any(|node| *node.name() == expected_name));
```
Comment (afck) [inline src/lib.rs]: That's much nicer! :+1:

File: src/lib.rs (hunk 1 above)
Comment (madadam) [inline src/lib.rs]: The purpose of that line (and the assert bellow) is to verify that no nodes were removed by adding a new node which falls into a bucket that is already full (Its in one of the bullet points in the "Changes to the routing logic" section in the RFC). So I only test that the number of nodes in the table increased by one. This test does not care about how many nodes there were initially.

Comment (madadam) [inline src/lib.rs]: On the other hand, asserting the exact number of nodes expected in the table would make the test slightly more readable, so I'll change it.

Comment (afck) [inline src/lib.rs]: :+1: 
And it never hurts to make the assertions stronger. :-)

**Pull request merged**